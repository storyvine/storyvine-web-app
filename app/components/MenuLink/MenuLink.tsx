
import * as React from 'react';
import { NavLink } from 'react-router-dom';
import { Icon } from 'antd';

interface Props {
  className?: string,
  name: string,
  iconName?: string,
  to: string,
  iconStyles?: Object,
  hideSidebar?: () => any,
};

const MenuLink = ({ className, to, name, iconName, iconStyles, hideSidebar }: Props) => (
  <NavLink to={to} className={className} onClick={hideSidebar}>
    {iconName && <Icon type={iconName} style={iconStyles} />}
    {name}
  </NavLink>
);

MenuLink.defaultProps = {
  iconStyles: null,
};

export default MenuLink;
