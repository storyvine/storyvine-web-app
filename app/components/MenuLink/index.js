// @flow
import React from 'react';
import { NavLink } from 'react-router-dom';
import { Icon } from 'antd';

type Props = {
  className: string,
  name: string,
  iconName: string,
  to: string,
  iconStyles?: Object | null,
};

const MenuLink = ({ className, to, name, iconName, iconStyles }: Props) => (
  <NavLink to={to} className={className}>
    <Icon type={iconName} style={iconStyles} />
    {name}
  </NavLink>
);

MenuLink.defaultProps = {
  iconStyles: null,
};

export default MenuLink;
