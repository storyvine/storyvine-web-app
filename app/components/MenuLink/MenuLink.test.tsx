import * as React from 'react';
import { shallow } from 'enzyme';
import MenuLink from './MenuLink';

test('MenuLink renders correctly', () => {
  expect(shallow(<MenuLink name="test" to="testPath" />)).toMatchSnapshot();
});

test('MenuLink renders correctly with icon', () => {
  expect(shallow(<MenuLink name="test" to="testPath" iconName="testName" />)).toMatchSnapshot();
});

test('MenuLink hide sidebar on click', () => {
  const hideSidebarSpy = jest.fn();
  const MenuLinkComponent = shallow(<MenuLink name="test" to="testPath" hideSidebar={hideSidebarSpy} />);
  MenuLinkComponent.find('NavLink').simulate('click');
  expect(hideSidebarSpy.mock.calls.length).toBe(1);
});
