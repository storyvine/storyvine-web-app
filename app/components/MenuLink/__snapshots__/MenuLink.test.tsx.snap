// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenuLink renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MenuLink
    iconStyles={null}
    name="test"
    to="testPath"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "activeClassName": "active",
      "aria-current": "page",
      "children": Array [
        undefined,
        "test",
      ],
      "className": undefined,
      "onClick": undefined,
      "to": "testPath",
    },
    "ref": null,
    "rendered": Array [
      undefined,
      "test",
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "activeClassName": "active",
        "aria-current": "page",
        "children": Array [
          undefined,
          "test",
        ],
        "className": undefined,
        "onClick": undefined,
        "to": "testPath",
      },
      "ref": null,
      "rendered": Array [
        undefined,
        "test",
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`MenuLink renders correctly with icon 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MenuLink
    iconName="testName"
    iconStyles={null}
    name="test"
    to="testPath"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "activeClassName": "active",
      "aria-current": "page",
      "children": Array [
        <Icon
          style={null}
          type="testName"
        />,
        "test",
      ],
      "className": undefined,
      "onClick": undefined,
      "to": "testPath",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "style": null,
          "type": "testName",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "test",
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "activeClassName": "active",
        "aria-current": "page",
        "children": Array [
          <Icon
            style={null}
            type="testName"
          />,
          "test",
        ],
        "className": undefined,
        "onClick": undefined,
        "to": "testPath",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "style": null,
            "type": "testName",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "test",
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
